main.go

```diff
@@ -128,77 +129,87 @@ func main() {
 
 	setupLog.V(0).Info("Configuration details", "Configuration", resourcemanagerconfig.ConfigString())
 
-	keyvaultName := resourcemanagerconfig.OperatorKeyvault()
+	keyvaultName := resourcemanagerconfig.GlobalCredentials().OperatorKeyvault()
 
 	if keyvaultName == "" {
 		setupLog.Info("Keyvault name is empty")
 		secretClient = k8sSecrets.New(mgr.GetClient())
 	} else {
 		setupLog.Info("Instantiating secrets client for keyvault " + keyvaultName)
-		secretClient = keyvaultSecrets.New(keyvaultName)
+		secretClient = keyvaultSecrets.New(keyvaultName, config.GlobalCredentials())
 	}
 
-	apimManager := resourceapimanagement.NewManager()
-	apimServiceManager := apimservice.NewAzureAPIMgmtServiceManager()
-	vnetManager := vnet.NewAzureVNetManager()
-	resourceGroupManager := resourcemanagerresourcegroup.NewAzureResourceGroupManager()
+	// TODO(creds-refactor): construction of these managers will need
```

Comment:
Yeah, I think this is a good change we should make - the small type-specific `*Reconciler` types should go away. This comment wasn't very clear because when I added it I didn't fully know where the managers were going to move to, but actually the construction will happen inside the `AsyncReconciler` rather than in the type-specific ones (that's the follow-up change I'm working on at the moment). I'll update it in the meantime.